service/controller/v3/resource/chart/desired.go

```diff
@@ -65,3 +73,47 @@ func (r *Resource) getConfigMapValues(ctx context.Context, customObject v1alpha1
 
 	return chartValues, nil
 }
+
+func (r *Resource) getSecretValues(ctx context.Context, customObject v1alpha1.ChartConfig) (map[string]interface{}, error) {
+	secretValues := make(map[string]interface{})
+
+	if key.SecretName(customObject) != "" {
```

Comment:
Why not quick return instead:

```go
if key.SecretName(customObject) == "" {
	return secretValues, nil
}
```

Reply:
I aligned this with the function for `configMaps` from ross for now. I will return early in both in a separate PR.